src/builtins/history.rs

```diff
@@ -343,6 +346,13 @@ pub fn history(parser: &Parser, streams: &mut IoStreams, args: &mut [&wstr]) ->
             history.save();
         }
         HistCmd::HIST_UNDEF => panic!("Unexpected HIST_UNDEF seen"),
+        HistCmd::HIST_PUSH =>
+        {
+            #[allow(clippy::unnecessary_to_owned)]
+            for arg in args.iter().copied() {
+                history.add_commandline(arg.to_owned());
+            }
```

Comment:
No need to silence the lint:

```suggestion
            for &arg in args {
                history.add_commandline(arg.to_owned());
            }
```

Reply:
I get this without silencing the lint, same as the `HIST_DELETE`

Reply:
Yes, if you follow the clippy suggestion and then fix the type error (oops, clippy bug), you get `for &arg in args` as in my suggestion.

Reply:
Oh sorry, I didn't notice the change, thanks. Will fix.